src/Particles/babylon.particleSystem.ts

```diff
@@ -865,11 +985,12 @@
          * Its lifetime will start back at 0.
          */
         public recycleParticle: (particle: Particle) => void = (particle) => {
-            var lastParticle = <Particle>this._particles.pop();
-            if (lastParticle !== particle) {
-                lastParticle.copyTo(particle);
+            // move particle from activeParticle list to stock particles
+            var index = this.particles.indexOf(particle);

```

Comment:
Note: The change for this section will remove instead of push/pop which may introduce a perf hit. 
this change was needed to ensure the same particle created is the one that gets disposed (I think it was a bug)

Reply:
Did perf test it was faster the other way. (tried 3 methods)

![image](https://user-images.githubusercontent.com/3753071/45122724-8d6f9900-b119-11e8-9b13-90e642b9dd67.png)


http://localhost:1338/Playground/index-local.html#6W42J5#7